### ISSUE TITLE
bug : influxdb metrics query editor - interactive - empty group by options

### ISSUE THREAD

**Issue by Daryes** [grafana/grafana#13861]
Grafana v5.3.2

Firefox v62 (win64)
Datasource : influxdb

reproduce steps : edit a graph panel, select the metrics tab, edit a query. 
Select a measurement, add a "group by" item. Add another "group by" item : the list is blank, no option are available, even LIMIT and SLIMIT are missing.
The panel need to be exited and reopened in edit mode to have the "group by" list filled.

Didn't have any problem with v4.6.3. 
For the record, the current Grafana installation uses a dedicated and empty directory, no elements of the previous version are accessible (aside the data dir).

**Comment by torkelo** [comment]
Very strange,  unable to replicate this in  Chrome or Firefox with v5.3.2 
https://play.grafana.org/d/AAy9r_bmk/influxdb-server-monitoring?panelId=1&fullscreen&edit&orgId=1&tab=metrics 

Any errors in the browser dev console ?

**Comment by Daryes** [comment]
It happens each time (Firefox 62.0.3), so I can test at will.

There are no noticeable errors at first glance in the dev console : 
note : I've cleared the domain name from the log.
````
22:37:42.430 window.controllers/Controllers is deprecated. Do not use it for UA detection. index.js:1362
22:37:43.093 warning rootScope onAppEvent called without localscope grafana_app.ts:86:10
22:37:43.524 PanelModel sizeChanged event and render events fired bootstrap.js:31
22:37:45.866 found diagramContainer diagramControl.js:644:4
22:37:45.866 Object {  } diagramControl.js:645:4
22:37:45.870 Data source query result invalid, missing data field:  Array [] metrics_panel_ctrl.ts:300:17
22:37:45.870 received data diagramControl.js:152:4
22:37:45.870 Array [] diagramControl.js:153:4
22:37:45.870 mapped dataList to series diagramControl.js:155:4
22:37:45.871 Array [] diagramControl.js:156:4
22:37:46.057 updating svg style diagramControl.js:692:6
22:38:24.763 [Show/hide message details.] XHRGET https://grafana.***INTERNAL DOMAIN***/api/datasources/proxy/2/query?db=telegraf&q=SHOW%20TAG%20KEYS%20FROM%20%22autogen%22.%22cpu%22%20WHERE%20%22cpu%22%20%3D%20%27cpu-total%27&epoch=ms [HTTP/1.1 200 OK 20ms]
````
No error or message in the console after the XHRGET.  
The influxdb data is even correctly present in the response to the XHRGET query.  

Seems the bug might be in the parsing of the returned data from influx VS the existing data in the interactive query builder.

**Comment by torkelo** [comment]
What version of Influxdb are you using ? Can you open chrome dev tools network tab and inspect the response from influxdb when you click the plus button to add a group by? It should fire a query to influxdb.

Do you get the same issue with the graph panel or only diagram plugin?

**Comment by Daryes** [comment]
Db version : influxdb-1.6.4_linux_amd64  

As said, with the Firefox dev tools, the response for the XHRGET query to populate the "group by" list has the expected data.  
Be it in the console or the network tab, as I've checked both, there are 9 tags in the DB, and the `SHOW TAG KEYS ...` XHRGET query returns the 9 expected tags in the dev tools.  

I've just done a test with chrome 70.0.3 : exact same problem.
Works as expected on the first change in the "group by", then the list stays empty on the other changes in the same "group by". This while the XHR query sent to influxdb returns all the expected tags.

The problem lies when parsing this data VS the data already in the query : it works correctly on the first tag change in the "group by", but will show an empty list on the second change and after in the "group by". Even if I delete some elements or completely clear the "group by" filters, the selection list will stay empty until the editor is closed/refreshed

Even if it's visible in the XHR query in the previous message, it might be useful (or not) to point the fact the influxdb datasource in Grafana is configured in proxy mode : Influxdb isn't directly reachable.

**Comment by torkelo** [comment]
Strange, i am on Chrome 70 as well and cannot replicate any issue with this dropdown. Could you replicate this on the dashboard I linked to? 

![image](https://user-images.githubusercontent.com/10999/47774370-258a9880-dced-11e8-9567-ba04f41ad681.png)

**Comment by Daryes** [comment]
> Could you replicate this on the dashboard I linked to?

Either a link is missing, or I don't understand what you're asking

**Comment by Daryes** [comment]
I'm separating from the previous message, because I finally found how it's happening.

1) the link you've provided in your first post isn't at the right version : the help is showing this : Grafana v5.3.0 (c613a31)

2) Even if I'm still using the v4.5.3 for now because of the reported bugs, I'm able to change version in a matter of seconds as long as the data backup is available. 
Previously, I've tested the v5.3.1 and the problem was already here. But after seeing your test version, I've downgraded to Grafana v5.3.0 : working perfectly. 
So, the bug appeared somewhere between Grafana v5.3.0 and the v5.3.1.

3) found the how it's happening : 
- screenshot of the browser inspector dev tool, focused on the "group by" button and the dropdown list on the first click.
![grafana_selector_try1](https://user-images.githubusercontent.com/14362813/47813642-fbb48e80-dd4b-11e8-881d-3c230df26cde.png)  

- now, on the second click after an item was selected. I loved seeing the "display:none" that wasn't changing like on the first click.
![grafana_selector_try2](https://user-images.githubusercontent.com/14362813/47813757-35859500-dd4c-11e8-82f8-2ee296b19d40.png)  


4) Do I need to fill an issue for the play test which is not at the latest version ? Joking :D
